tests/vm_networking_test.go

```diff
@@ -0,0 +1,189 @@
+			_, err = expecter.ExpectBatch([]expect.Batcher{
+				&expect.BSnd{S: "\n"},
+				&expect.BExp{R: "\\$ "},
+				&expect.BSnd{S: "nc -klp 1500 -e echo -e \"Hello World!\"\n"},
+			}, 60*time.Second)
+			Expect(err).ToNot(HaveOccurred())
+		}()
+
+		// Create a VM and log in, to allow executing arbitrary commands from the terminal
+		go func() {
+			defer wg.Done()
+			defer GinkgoRecover()
+			outboundVM = createAndLogin()
+		}()
+
+		wg.Wait()
+	})
+
+	Context("VirtualMachine with nodeNetwork definition given", func() {
+
+		It("should be able to reach the internet", func() {
+			// Wait until the VM is booted, ping google and check if we can reach the internet
+			expecter, _, err := tests.NewConsoleExpecter(virtClient, outboundVM, "", 10*time.Second)
+			defer expecter.Close()
+			Expect(err).ToNot(HaveOccurred())
+			_, err = expecter.ExpectBatch([]expect.Batcher{
+				&expect.BSnd{S: "\n"},
+				&expect.BExp{R: "\\$ "},
+				&expect.BSnd{S: "ping www.google.com -c 1 -w 5\n"},
+				&expect.BExp{R: "\\$ "},
+				&expect.BSnd{S: "echo $?\n"},
+				&expect.BExp{R: "0"},
+			}, 90*time.Second)
+			Expect(err).ToNot(HaveOccurred())
+		}, 120)
+
+		table.DescribeTable("should be reachable via the propagated IP from a Pod", func(op v12.NodeSelectorOperator, hostNetwork bool) {
+
+			ip := inboundVM.Status.Interfaces[0].IP
+
+			// Run netcat and give it one second to ghet "Hello World!" back from the VM
+			check := []string{fmt.Sprintf("while read x; do test \"$x\" = \"Hello World!\"; exit $?; done < <(nc %s 1500 -i 1 -w 1)", ip)}
+			job := tests.RenderJob("netcat", dockerTag, []string{"/bin/bash", "-c"}, check)
+			job.Spec.Affinity = &v12.Affinity{
+				NodeAffinity: &v12.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &v12.NodeSelector{
+						NodeSelectorTerms: []v12.NodeSelectorTerm{
+							{
+								MatchExpressions: []v12.NodeSelectorRequirement{
+									{Key: "kubernetes.io/hostname", Operator: op, Values: []string{inboundVM.Status.NodeName}},
+								},
+							},
+						},
+					},
+				},
+			}
+			job.Spec.HostNetwork = hostNetwork
+
+			job, err = virtClient.CoreV1().Pods(inboundVM.ObjectMeta.Namespace).Create(job)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() v12.PodPhase {
+				j, err := virtClient.Core().Pods(inboundVM.ObjectMeta.Namespace).Get(job.ObjectMeta.Name, v13.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(j.Status.Phase).ToNot(Equal(v12.PodFailed))
+				return j.Status.Phase
+			}, 30*time.Second, 1*time.Second).Should(Equal(v12.PodSucceeded))
+
+		},
+			table.Entry("on the same node from Pod", v12.NodeSelectorOpIn, false),
+			table.Entry("on a different node from Pod", v12.NodeSelectorOpNotIn, false),
+			table.Entry("on the same node from Node", v12.NodeSelectorOpIn, true),
+			table.Entry("on a different node from Node", v12.NodeSelectorOpNotIn, true),
+		)
+	})
+
+})
+
+func BeforeAll(fn func()) {
+	first := true
+	BeforeEach(func() {
```

Comment:
Ha, that's one way to do it. I guess ginkgo doesn't have a BeforeAll() function.  I like the idea of setting up the environment in parallel before the tests start. It cuts down on test time quite a bit.